internal/controller/apiextensions/composite/composed.go

```diff
@@ -159,7 +160,7 @@ func NewAPIConnectionDetailsFetcher(c client.Client) *APIConnectionDetailsFetche
 }
 
 // FetchConnectionDetails of the supplied composed resource, if any.
-func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) {
+func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) { // nolint:gocyclo
 	sref := cd.GetWriteConnectionSecretToReference()
 	if sref == nil {
```

Comment:
When using `fromFieldPath`, does `writeConnectionSecretToRef` still required on the managed resource? Wouldn't it be reasonable to expect `writeConnectionSecretToRef` only needed when using with `fromConnectionSecretKey`

Reply:
I would think writeConnectionSecretToRef is still required since we still need to write the values to a specific secret.

Reply:
In my experiment, crossplane will write the values to a working connection secret with uid name to the namespace specified in `writeConnectionSecretsToNamespace`. I don't need to specify `writeConnectionSecretToRef` when `fromConnectionSecretKey` is not used. Same goes to fixed `name` and `value` pair

Reply:
Ahh yes, I am not sure, let me test out this behavior

Reply:
I understand now, I think it would be okay to make the GetWriteConnectionSecretToReference optional, this way we can still publish values/FromResourceFieldPath even if there is no secret for a resource.

How does that sound to you @negz?

Reply:
That sounds good to me!